doc/source/getting_started/installing.rst

```diff
@@ -0,0 +1,458 @@
+- ``pywavelets`` : Install `PyWavelets`_ for wavelet transforms.
+- ``scikit`` : Install `scikit-image`_ as a simple backend for ray transforms.
+- ``proximal``: Install the `ProxImaL`_ convex optimization package.
+
+
+These dependencies are optional and may not be easy to install on your system (especially on Windows).
+In general, a clean ODL installation is enough for most users' initial needs.
+
+
+------
+
+
+********************************************************************
+Alternative 3: Installation from source (recommended for developers)
+********************************************************************
+This installation method is intended for developers who want to make changes to the code.
+It assumes that the `Git`_ version control system is available on your system; for up-to-date instructions, consult the `Git installation instructions <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>`_.
+You also need `pip`_ to perform the installation.
+
+.. note::
+    You should consider performing all described steps in a `conda environment <http://conda.pydata.org/docs/using/envs.html>`_ -- it gives you the same encapsulation benefits as developer that you would enjoy also as a user (no conflicting packages, free to choose Python version, ...).
+    See the `Installing Anaconda`_ section for setup instructions.
+
+To get ODL, clone the repository with the command
+
+.. code-block:: bash
+
+   $ git clone https://github.com/odlgroup/odl
+
+No GitHub account is required for this step.
+
+
+In a conda environment
+======================
+You can choose to install dependencies first (optional ones in square brackets):
+
+.. code-block:: bash
+
+    $ conda install nomkl numpy scipy future [matplotlib]
+
+After that, enter the top-level directory of the cloned repository and run
+
+.. code-block:: bash
+
+   $ pip install --editable .
+
+Using only ``pip``
+==================
+Enter the top-level directory of the cloned repository and run
+
+.. code-block:: bash
+
+   $ pip install --user --editable .
+
+
+.. warning::
+    **Don't forget the "." (dot) at the end** - it refers to the current directory, the location from where ``pip`` is supposed to install ODL.
+
+.. note::
+    We recommend the ``--editable`` option (can be shortened to ``-e``) since it installs a link instead of copying the files to your Python packages location.
+    This way, local changes to the code (e.g. after a ``git pull``) take immediate effect without reinstallation.
+
+
+Further developer information
+=============================
+See :ref:`Contributing to ODL <contributing>` for more information.
+
+
+------
+
+
+****************
+Runing the tests
+****************
+Unit tests in ODL are based on `pytest`_, and coverage reports are created by the `coverage`_ module.
+
+Installing testing dependencies
+===============================
+
+Using ``conda``:
+----------------
+.. code-block:: bash
+
+    $ conda install pytest pytest-cov pytest-pep8
```

Comment:
Well this line of code is independent of ODL, it doesnt mention it anywhere. Hence updating our dependencies would not make any difference here. Checking in code is indeed an option (that we should do), but it would only lead to users running this, getting an error and then having to run update anyway.